packages/gated-content/src/conditions/advanced-contract-condition.ts

```diff
@@ -0,0 +1,150 @@
+  assertValidComparison(condition);
+
+  return [
+    {
+      conditionType: LitConditionType.EVM_CONTRACT,
+      contractAddress: condition.contract.address.toLowerCase(),
+      chain: toLitSupportedChainName(condition.contract.chainId),
+      functionName: condition.functionName,
+      functionParams: condition.params || [],
+      functionAbi: new Interface([condition.abi]).getFunction(condition.functionName),
+      returnValueTest: {
+        key: '',
+        comparator: resolveScalarOperatorSymbol(condition.comparison),
+        value: condition.value,
+      },
+    },
+  ];
+};
+
+function assertValidAbi(humanReadableAbi: string, functionName: string): void {
+  try {
+    const fn = new Interface([humanReadableAbi]).getFunction(functionName);
+
+    // assert view function
+    invariant(fn.stateMutability === 'view', 'unsupported');
+
+    // assert single output
+    invariant(Array.isArray(fn.outputs) && fn.outputs.length === 1, 'unsupported');
+
+    // assert output is boolean or uint
+    invariant(
+      fn.outputs[0] && (fn.outputs[0].type === 'bool' || fn.outputs[0].type === 'uint256'),
+      'unsupported',
+    );
+  } catch (e: unknown) {
+    throw new InvalidAccessCriteriaError(
+      `Invalid abi: ${humanReadableAbi} or function: ${functionName}. Only view functions returning a single boolean or uint output are supported`,
+    );
+  }
+}
+
+/**
+ * verifies arguments are valid, as well as the prefixed `:userAddress` param exists
+ * @param condition - the user provided condition object
+ */
+function assertValidFunctionParams(condition: AdvancedContractCondition): void {
+  try {
+    const fn = new Interface([condition.abi]).getFunction(condition.functionName);
+    let userAddressParamFound = false;
+
+    invariant(fn.inputs.length === condition.params.length, 'wrong number of params');
+
+    fn.inputs.forEach((input, index) => {
+      const param = condition.params[index];
+
+      invariant(param, `param ${input.name || input.type} is missing`);
+
+      if (input.baseType === 'array' || input.baseType === 'tuple') {
+        invariant(
+          Array.isArray(param) && param.length > 0,
+          `param ${input.name} expects an array argument`,
+        );
+
+        if (param.includes(':userAddress')) {
+          userAddressParamFound = true;
+        }
+      }
+
+      if (input.baseType === 'address') {
+        if (param === ':userAddress') {
+          userAddressParamFound = true;
+        } else {
+          assertValidAddress(param);
+        }
+      } else if (input.baseType.includes('int')) {
+        BigNumber.from(param);
+      } else if (input.baseType === 'bool') {
+        invariant(
+          param === 'true' || param === 'false',
+          `param ${input.name} is invalid, must be a boolean)`,
+        );
+      } else if (input.baseType === 'bytes') {
+        invariant(param.startsWith('0x'), `param ${input.name} is invalid, must be a hex string)`);
```

Comment:
a bit stronger checks 
```suggestion
        invariant(/^0x[0-9A-Fa-f]+$/.test(param), `param ${input.name} is invalid, must be a hex string)`);
```